# Goal Split Subgroup: Correct Resolution Model

The following models takes into account that, throughout analyses, *no_goal* seems to behave entirely differently to the other two Goal Type conditions (*goal_frequent* and *goal_non_frequent*). To do so, it splits the data accordingly and looks into goals on the one hand and no_goals on the other separately.

Import Libraries

In [180]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## Goal Split Subgroup

1. Create resolved_correct (goals and no_goals)
2. Counts and Proportions (goals and no_goals)
3. Logistic Models

    - Goals

        Main Effect Models:

        - RC01: Focus
        - RC02: Agent
        - RC03: Goal_Type

        Additive Models:

        - RC04: Focus + Agent
        - RC05: Focus + Goal_Type
        - RC06: Agent + Goal_Type

        Interactive Models:

        - RC07: Focus * Agent
        - RC08: Focus * Goal_Type
        - RC09: Agent * Goal_Type
        - RC10: Focus * Agent + Goal_Type

    - No_goals

        Main Effect Models:

        - RC11:
        - RC12:

        Additive Model:

        - RC13:

        Interactive Model:

        - RC14:

4. Model Comparisons
5. Summary Table

Read Data

In [181]:
subgroup_theoretical = pd.read_csv("../../data/processed/subgroup_theoretical.csv")

1. Create resolved_correct

Binary Outcome: correct responses from escape_L2 observations.

In [182]:
subgroup_theoretical["escape_L2"] = (
    subgroup_theoretical["Response_Full"] != "L2_other"
).astype(int)

In [183]:
escapees = subgroup_theoretical[subgroup_theoretical["escape_L2"] == 1].copy()

In [184]:
escapees["resolved_correct"] = (escapees["Response_Full"] == "correct").astype(int)

Split escapees data into goals vs. no_goals

In [185]:
goals_escapees = escapees[escapees["Goal_Type"] != "no_goal"].copy()

In [186]:
no_goal_escapees = escapees[escapees["Goal_Type"] == "no_goal"].copy()

2. Counts and Proportions for goals and no_goals

General resolved_correct reponses count:

In [187]:
resolved_correct_counts = pd.crosstab(
    escapees["resolved_correct"],
    escapees["Response_Full"],
    margins= True)

resolved_correct_counts

Response_Full,L1_transfer,correct,missing_response,All
resolved_correct,,,,
0,96,0,3,99
1,0,61,0,61
All,96,61,3,160


resolved_correct responses count for goals:

In [188]:
goals_resolved_correct_counts = pd.crosstab(
    goals_escapees["resolved_correct"],
    goals_escapees["Response_Full"],
    margins= True)

goals_resolved_correct_counts

Response_Full,L1_transfer,correct,missing_response,All
resolved_correct,,,,
0,62,0,1,63
1,0,43,0,43
All,62,43,1,106


resolved_correct responses count for no_goals:

In [189]:
no_goal_resolved_correct_counts = pd.crosstab(
    no_goal_escapees["resolved_correct"],
    no_goal_escapees["Response_Full"],
    margins= True)

no_goal_resolved_correct_counts

Response_Full,L1_transfer,correct,missing_response,All
resolved_correct,,,,
0,34,0,2,36
1,0,18,0,18
All,34,18,2,54


General resolved_correct responses proportions:

In [190]:
resolved_correct_props = escapees["resolved_correct"].value_counts(normalize = True)
resolved_correct_props

resolved_correct
0    0.61875
1    0.38125
Name: proportion, dtype: float64

From 160 observations, almost 38% of responses that had escaped *L2_other* options were resolved correctly. 

resolved_correct responses proportions for goals:

In [191]:
goals_resolved_correct_props = goals_escapees["resolved_correct"].value_counts(normalize=True)
goals_resolved_correct_props

resolved_correct
0    0.59434
1    0.40566
Name: proportion, dtype: float64

From 106 observations, 40.5% of goal responses that had escaped *L2_other* options were resolved correctly. 

resolved_correct responses proportions for no_goals:

In [192]:
no_goal_resolved_correct_props = no_goal_escapees["resolved_correct"].value_counts(normalize=True)
no_goal_resolved_correct_props

resolved_correct
0    0.666667
1    0.333333
Name: proportion, dtype: float64

From 54 observations, 33% of no_goal responses that had escaped *L2_other* options were resolved correctly. 

- The proportion distribution of correct resolution responses is higher when there is a goal oriented representation (40.5%) as opposed to when there isn't a representation of a goal (33.3%). 

Counts and Proportions by Condition:

In [193]:
counts_condition = pd.crosstab(
    [escapees["Goal_Type"], escapees["Agent"],escapees["Focus"]],
    escapees["resolved_correct"],
    margins = True)

counts_condition

resolved_correct                0   1  All
Goal_Type         Agent Focus             
goal_frequent     0     I       7   9   16
                        They    7   5   12
                  1     I       7   7   14
                        They    5   3    8
goal_non_frequent 0     I       7   9   16
                        They   14   3   17
                  1     I      11   4   15
                        They    5   3    8
no_goal           0     I      15   6   21
                        They   10   3   13
                  1     I       9   5   14
                        They    2   4    6
All                            99  61  160

In [194]:
goals_counts_condition = pd.crosstab(
    [goals_escapees["Goal_Type"], goals_escapees["Agent"], goals_escapees["Focus"]],
    goals_escapees["resolved_correct"],
    margins = True)

goals_counts_condition

resolved_correct                0   1  All
Goal_Type         Agent Focus             
goal_frequent     0     I       7   9   16
                        They    7   5   12
                  1     I       7   7   14
                        They    5   3    8
goal_non_frequent 0     I       7   9   16
                        They   14   3   17
                  1     I      11   4   15
                        They    5   3    8
All                            63  43  106

In [195]:
no_goal_counts_condition = pd.crosstab(
    [no_goal_escapees["Goal_Type"], no_goal_escapees["Agent"], no_goal_escapees["Focus"]],
    no_goal_escapees["resolved_correct"],
    margins = True)

no_goal_counts_condition

resolved_correct        0   1  All
Goal_Type Agent Focus             
no_goal   0     I      15   6   21
                They   10   3   13
          1     I       9   5   14
                They    2   4    6
All                    36  18   54

In [196]:
props_condition = pd.crosstab(
    [escapees["Goal_Type"], escapees["Agent"], escapees["Focus"]],
    escapees["resolved_correct"],
    normalize= "index")

props_condition

resolved_correct                      0         1
Goal_Type         Agent Focus                    
goal_frequent     0     I      0.437500  0.562500
                        They   0.583333  0.416667
                  1     I      0.500000  0.500000
                        They   0.625000  0.375000
goal_non_frequent 0     I      0.437500  0.562500
                        They   0.823529  0.176471
                  1     I      0.733333  0.266667
                        They   0.625000  0.375000
no_goal           0     I      0.714286  0.285714
                        They   0.769231  0.230769
                  1     I      0.642857  0.357143
                        They   0.333333  0.666667

In [197]:
goals_props_condition = pd.crosstab(
    [goals_escapees["Goal_Type"], goals_escapees["Agent"], goals_escapees["Focus"]],
    goals_escapees["resolved_correct"],
    normalize= "index")

goals_props_condition

resolved_correct                      0         1
Goal_Type         Agent Focus                    
goal_frequent     0     I      0.437500  0.562500
                        They   0.583333  0.416667
                  1     I      0.500000  0.500000
                        They   0.625000  0.375000
goal_non_frequent 0     I      0.437500  0.562500
                        They   0.823529  0.176471
                  1     I      0.733333  0.266667
                        They   0.625000  0.375000

In [198]:
no_goal_props_condition = pd.crosstab(
    [no_goal_escapees["Goal_Type"], no_goal_escapees["Agent"], no_goal_escapees["Focus"]],
    no_goal_escapees["resolved_correct"],
    normalize= "index")

no_goal_props_condition

resolved_correct              0         1
Goal_Type Agent Focus                    
no_goal   0     I      0.714286  0.285714
                They   0.769231  0.230769
          1     I      0.642857  0.357143
                They   0.333333  0.666667

COMMENT GENERAL PROPORTIONS

3. Logistic Models

Import Libraries

In [199]:
import statsmodels.formula.api as smf
from scipy.stats import chi2

Sanity checks:

In [200]:
escapees["resolved_correct"].value_counts()

resolved_correct
0    99
1    61
Name: count, dtype: int64

In [201]:
goals_escapees["resolved_correct"].value_counts()

resolved_correct
0    63
1    43
Name: count, dtype: int64

In [202]:
no_goal_escapees["resolved_correct"].value_counts()

resolved_correct
0    36
1    18
Name: count, dtype: int64

**Goals**

**Main Effect Models**

**RC01**

resolved_correct ~ Focus

In [203]:
RC01 = smf.logit(
    "resolved_correct ~ Focus",
    data=goals_escapees
    ).fit()

print(RC01.summary())

Optimization terminated successfully.
         Current function value: 0.661392
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:       resolved_correct   No. Observations:                  106
Model:                          Logit   Df Residuals:                      104
Method:                           MLE   Df Model:                            1
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.02051
Time:                        15:45:35   Log-Likelihood:                -70.108
converged:                       True   LL-Null:                       -71.575
Covariance Type:            nonrobust   LLR p-value:                   0.08664
                    coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept        -0.0984      0.256     -0.384      0.701      -0.601       0.404
Focus[T.They]    -0.

**RC02**

resolved_correct ~ Agent

In [204]:
RC02 = smf.logit(
    "resolved_correct ~ Agent",
    data=goals_escapees
    ).fit()

print(RC02.summary())

Optimization terminated successfully.
         Current function value: 0.674048
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:       resolved_correct   No. Observations:                  106
Model:                          Logit   Df Residuals:                      104
Method:                           MLE   Df Model:                            1
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                0.001766
Time:                        15:45:35   Log-Likelihood:                -71.449
converged:                       True   LL-Null:                       -71.575
Covariance Type:            nonrobust   LLR p-value:                    0.6151
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -0.2973      0.259     -1.148      0.251      -0.805       0.210
Agent         -0.2017      0.

**RC03**

resolved_correct ~ Goal_Type

In [205]:
RC03 = smf.logit(
    "resolved_correct ~ Goal_Type",
    data=goals_escapees
    ).fit()

print(RC03.summary())

Optimization terminated successfully.
         Current function value: 0.664988
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:       resolved_correct   No. Observations:                  106
Model:                          Logit   Df Residuals:                      104
Method:                           MLE   Df Model:                            1
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.01518
Time:                        15:45:35   Log-Likelihood:                -70.489
converged:                       True   LL-Null:                       -71.575
Covariance Type:            nonrobust   LLR p-value:                    0.1404
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                         -0.0800      0.283     -0.283     

**Additive Models**

**RC04**

resolved_correct ~ Focus + Agent

In [206]:
RC04 = smf.logit(
    "resolved_correct ~ Focus + Agent",
    data=goals_escapees
    ).fit()

print(RC04.summary())

Optimization terminated successfully.
         Current function value: 0.658933
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:       resolved_correct   No. Observations:                  106
Model:                          Logit   Df Residuals:                      103
Method:                           MLE   Df Model:                            2
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.02415
Time:                        15:45:35   Log-Likelihood:                -69.847
converged:                       True   LL-Null:                       -71.575
Covariance Type:            nonrobust   LLR p-value:                    0.1775
                    coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept         0.0419      0.322      0.130      0.897      -0.590       0.673
Focus[T.They]    -0.

**RC05**

resolved_correct ~ Focus + Goal_Type

In [207]:
RC05 = smf.logit(
    "resolved_correct ~ Focus + Goal_Type",
    data=goals_escapees
    ).fit()

print(RC05.summary())

Optimization terminated successfully.
         Current function value: 0.651957
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:       resolved_correct   No. Observations:                  106
Model:                          Logit   Df Residuals:                      103
Method:                           MLE   Df Model:                            2
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.03448
Time:                        15:45:35   Log-Likelihood:                -69.107
converged:                       True   LL-Null:                       -71.575
Covariance Type:            nonrobust   LLR p-value:                   0.08476
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                          0.1894      0.329      0.576     

**RC06**

resolved_correct ~  Agent + Goal_Type

In [208]:
RC06 = smf.logit(
    "resolved_correct ~ Agent + Goal_Type",
    data=goals_escapees
    ).fit()

print(RC06.summary())

Optimization terminated successfully.
         Current function value: 0.663549
         Iterations 4
                           Logit Regression Results                           
Dep. Variable:       resolved_correct   No. Observations:                  106
Model:                          Logit   Df Residuals:                      103
Method:                           MLE   Df Model:                            2
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.01731
Time:                        15:45:35   Log-Likelihood:                -70.336
converged:                       True   LL-Null:                       -71.575
Covariance Type:            nonrobust   LLR p-value:                    0.2896
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                          0.0183      0.335      0.055     

**Interactive Models**

**RC07**

resolved_correct ~ Focus * Agent

In [209]:
RC07 = smf.logit(
    "resolved_correct ~ Focus * Agent",
    data=goals_escapees
    ).fit()

print(RC07.summary())

Optimization terminated successfully.
         Current function value: 0.649473
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:       resolved_correct   No. Observations:                  106
Model:                          Logit   Df Residuals:                      102
Method:                           MLE   Df Model:                            3
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.03816
Time:                        15:45:35   Log-Likelihood:                -68.844
converged:                       True   LL-Null:                       -71.575
Covariance Type:            nonrobust   LLR p-value:                    0.1409
                          coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------------
Intercept               0.2513      0.356      0.705      0.481      -0.447       0.950
Fo

**RC08**

resolved_correct ~ Focus * Goal_Type

In [210]:
RC08 = smf.logit(
    "resolved_correct ~ Focus * Goal_Type",
    data=goals_escapees
    ).fit()

print(RC08.summary())

Optimization terminated successfully.
         Current function value: 0.651392
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:       resolved_correct   No. Observations:                  106
Model:                          Logit   Df Residuals:                      102
Method:                           MLE   Df Model:                            3
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.03532
Time:                        15:45:35   Log-Likelihood:                -69.048
converged:                       True   LL-Null:                       -71.575
Covariance Type:            nonrobust   LLR p-value:                    0.1677
                                                   coef    std err          z      P>|z|      [0.025      0.975]
----------------------------------------------------------------------------------------------------------------
Intercept                               

**RC09**

resolved_correct ~ Agent * Goal_Type

In [211]:
RC09 = smf.logit(
    "resolved_correct ~ Agent * Goal_Type",
    data=goals_escapees
    ).fit()

print(RC09.summary())

Optimization terminated successfully.
         Current function value: 0.663498
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:       resolved_correct   No. Observations:                  106
Model:                          Logit   Df Residuals:                      102
Method:                           MLE   Df Model:                            3
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.01739
Time:                        15:45:35   Log-Likelihood:                -70.331
converged:                       True   LL-Null:                       -71.575
Covariance Type:            nonrobust   LLR p-value:                    0.4772
                                           coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------------
Intercept                            -1.226e-16      0.3

**RC10**

resolved_correct ~ Focus * Agent + Goal_Type

In [212]:
RC10 = smf.logit(
    "resolved_correct ~ Focus * Agent + Goal_Type",
    data=goals_escapees
    ).fit()

print(RC10.summary())

Optimization terminated successfully.
         Current function value: 0.640518
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:       resolved_correct   No. Observations:                  106
Model:                          Logit   Df Residuals:                      101
Method:                           MLE   Df Model:                            4
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.05142
Time:                        15:45:35   Log-Likelihood:                -67.895
converged:                       True   LL-Null:                       -71.575
Covariance Type:            nonrobust   LLR p-value:                    0.1180
                                     coef    std err          z      P>|z|      [0.025      0.975]
--------------------------------------------------------------------------------------------------
Intercept                          0.5378      0.418      1.287     

**No Goals**

**Main Effect Models**

**RC11**

resolved_correct ~ Focus 

In [213]:
RC11 = smf.logit(
    "resolved_correct ~ Focus",
    data=no_goal_escapees
    ).fit()

print(RC11.summary())

Optimization terminated successfully.
         Current function value: 0.635021
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:       resolved_correct   No. Observations:                   54
Model:                          Logit   Df Residuals:                       52
Method:                           MLE   Df Model:                            1
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                0.002346
Time:                        15:45:35   Log-Likelihood:                -34.291
converged:                       True   LL-Null:                       -34.372
Covariance Type:            nonrobust   LLR p-value:                    0.6880
                    coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept        -0.7802      0.364     -2.143      0.032      -1.494      -0.067
Focus[T.They]     0.

**RC12**

resolved_correct ~ Agent 

In [214]:
RC12 = smf.logit(
    "resolved_correct ~ Agent",
    data=no_goal_escapees
    ).fit()

print(RC12.summary())

Optimization terminated successfully.
         Current function value: 0.618743
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:       resolved_correct   No. Observations:                   54
Model:                          Logit   Df Residuals:                       52
Method:                           MLE   Df Model:                            1
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.02792
Time:                        15:45:35   Log-Likelihood:                -33.412
converged:                       True   LL-Null:                       -34.372
Covariance Type:            nonrobust   LLR p-value:                    0.1659
                 coef    std err          z      P>|z|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept     -1.0217      0.389     -2.628      0.009      -1.784      -0.260
Agent          0.8210      0.

**Additive Model**

**RC13**

resolved_correct ~ Focus + Agent

In [215]:
RC13 = smf.logit(
    "resolved_correct ~ Focus + Agent",
    data=no_goal_escapees
    ).fit()

print(RC13.summary())

Optimization terminated successfully.
         Current function value: 0.616137
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:       resolved_correct   No. Observations:                   54
Model:                          Logit   Df Residuals:                       51
Method:                           MLE   Df Model:                            2
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.03201
Time:                        15:45:35   Log-Likelihood:                -33.271
converged:                       True   LL-Null:                       -34.372
Covariance Type:            nonrobust   LLR p-value:                    0.3328
                    coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------
Intercept        -1.1525      0.467     -2.468      0.014      -2.068      -0.237
Focus[T.They]     0.

**Interactive Model**

**RC14**

resolved_correct ~ Focus * Agent 

In [216]:
RC14 = smf.logit(
    "resolved_correct ~ Focus * Agent",
    data=no_goal_escapees
    ).fit()

print(RC14.summary())

Optimization terminated successfully.
         Current function value: 0.602407
         Iterations 5
                           Logit Regression Results                           
Dep. Variable:       resolved_correct   No. Observations:                   54
Model:                          Logit   Df Residuals:                       50
Method:                           MLE   Df Model:                            3
Date:                Fri, 17 Jul 2026   Pseudo R-squ.:                 0.05358
Time:                        15:45:35   Log-Likelihood:                -32.530
converged:                       True   LL-Null:                       -34.372
Covariance Type:            nonrobust   LLR p-value:                    0.2977
                          coef    std err          z      P>|z|      [0.025      0.975]
---------------------------------------------------------------------------------------
Intercept              -0.9163      0.483     -1.897      0.058      -1.863       0.030
Fo

4. Model Comparisons

In [217]:
from scipy.stats import chi2

def lr_test(model_small, model_large):

    LR = 2 * (model_large.llf - model_small.llf)

    df = model_large.df_model - model_small.df_model

    p = chi2.sf(LR, df)

    return pd.Series({
        "LR": LR,
        "df": df,
        "p_value": p
    })

In [218]:
lr_test(RC01, RC03)

LR        -0.762312
df         0.000000
p_value         NaN
dtype: float64

In [219]:
lr_test(RC01, RC04)

LR         0.521372
df         1.000000
p_value    0.470257
dtype: float64

5. Summary Table

In [220]:
goal_behaviour = ["Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "Goal", "No_goal", "No_goal", "No_goal", "No_goal", "No_goal", "No_goal", "No_goal"]

model_name = ["RC01", "RC02", "RC03", "RC04", "RC04", "RC05", "RC05", "RC06", "RC06", "RC07", "RC07", "RC07", "RC08", "RC08", "RC08", "RC09", "RC09", "RC09", "RC10", "RC10", "RC10", "RC10", "RC11", "RC12", "RC13", "RC13", "RC14", "RC14", "RC14"]

outcome = ["resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct", "resolved_correct"]

formula = ["esolved_correct ~ Focus", "esolved_correct ~ Agent", "esolved_correct ~ Goal_Type", "esolved_correct ~ Focus + Agent", "esolved_correct ~ Focus + Agent", "esolved_correct ~ Focus + Goal_Type", "esolved_correct ~ Focus + Goal_Type", "esolved_correct ~ Agent + Goal_Type", "esolved_correct ~ Agent + Goal_Type", "esolved_correct ~ Focus * Agent", "esolved_correct ~ Focus * Agent", "esolved_correct ~ Focus * Agent", "esolved_correct ~ Focus * Goal_Type", "esolved_correct ~ Focus * Goal_Type", "esolved_correct ~ Focus * Goal_Type", "esolved_correct ~ Agent * Goal_Type", "esolved_correct ~ Agent * Goal_Type", "esolved_correct ~ Agent * Goal_Type", "esolved_correct ~ Focus * Agent + Goal_Type", "esolved_correct ~ Focus * Agent + Goal_Type", "esolved_correct ~ Focus * Agent + Goal_Type", "esolved_correct ~ Focus * Agent + Goal_Type", "esolved_correct ~ Focus", "esolved_correct ~ Agent", "esolved_correct ~ Focus + Agent", "esolved_correct ~ Focus + Agent", "esolved_correct ~ Focus * Agent", "esolved_correct ~ Focus * Agent", "esolved_correct ~ Focus * Agent" ]

predictor= ["They", "Agent", "goal_non_frequent", "They", "Agent", "They", "goal_non_frequent", "goal_non_frequent", "Agent", "They", "Agent", "They:Agent", "They", "goal_non_frequent", "They:goal_non_frequent", "goal_non_frequent", "Agent", "Agent:goal_non_frequent", "They", "goal_non_frequent", "Agent", "They:Agent", "They", "Agent", "They", "Agent", "They", "Agent", "They:Agent"]

coef = ["RC01", -0.2017, -0.5864, "RC04", "RC04", "RC05", "RC05", -0.5947, -0.2241, "RC07", "RC07", "RC07", "RC08", "RC08", "RC08", -0.5596, -0.1823, -0.0847, "RC10", "RC10", "RC10", "RC10", "RC11", 0.8210, "RC13", "RC13", "RC14", "RC14", "RC14"]

std_error = ["RC01", 0.402, 0.400, "RC04", "RC04", "RC05", "RC05", 0.401, 0.407, "RC07", "RC07", "RC07", "RC08", "RC08", "RC08", 0.523, 0.571, 0.814, "RC10", "RC10", "RC10", "RC10", "RC11", 0.594, "RC13", "RC13", "RC14", "RC14", "RC14"]

coef_p = ["RC01", 0.616, 0.142, "RC04", "RC04", "RC05", "RC05", 0.138, 0.582, "RC07", "RC07", "RC07", "RC08", "RC08", "RC08", 0.285, 0.750, 0.917, "RC10", "RC10", "RC10", "RC10", "RC11", 0.167, "RC13", "RC13", "RC14", "RC14", "RC14"]

LLR_p = ["RC01", 0.6151, 0.1404, "RC04", "RC04", "RC05", "RC05", 0.2896, 0.2896, "RC07", "RC07", "RC07", "RC08", "RC08", "RC08", 0.4772, 0.4772, 0.4772, "RC10", "RC10", "RC10", "RC10", "RC11", 0.1659, "RC13", "RC13", "RC14", "RC14", "RC14"]

observations= [106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 106, 54, 54, 54, 54, 54, 54, 54]

goal_split_correct_summary = pd.DataFrame({
    "Goal_Behaviour": goal_behaviour,
    "Model": model_name,
    "Outcome": outcome,
    "Formula": formula, 
    "Predictor": predictor,
    "Coef": coef,
    "Std_err": std_error,
    "Coef_p": coef_p,
    "Model_LLR_p": LLR_p,
    "N": observations,
})

goal_split_correct_summary